src/rules/noWarningCommentsRule.ts

```diff
@@ -0,0 +1,163 @@
+    }
+}
+
+function parseOptions(options: IOptionsObject): Options {
+    if (options && options.location !== "start" && options.location !== "anywhere") {
+        throw new Error(`Option 'location' value is invalid: ${options.location}`);
+    }
+
+    if (options && !(options.terms instanceof Array)) {
+        throw new Error("Option 'terms' must be a string array.");
+    }
+
+    return {
+        location: options && options.location ? options.location as "start" | "anywhere" : "start",
+        terms: options && options.terms ? options.terms : ["todo", "fixme", "xxx"],
+    };
+}
+
+function checkCommentContainsWarningTerms(comment: string, reTerm: RegExpWithTerm[], rng: ts.CommentRange, ctx: Lint.WalkContext<Options>) {
+    reTerm.forEach((regexpWithTerm) => {
+        const match = regexpWithTerm.regexp.exec(comment);
+        if (match) {
+            ctx.addFailure(rng.pos, rng.end, Rule.COMMENT_ERROR_FACTORY(regexpWithTerm.term, ctx.options.location));
+        }
+    });
+}
+
+function convertToRegExp(term: string, location: "start" | "anywhere") {
+    const escaped = term.replace(/[-/\\$^*+?.()|[\]{}]/g, "\\$&");
+    let prefix;
+
+    /*
+     * If the term ends in a word character (a-z0-9_), ensure a word
+     * boundary at the end, so that substrings do not get falsely
+     * matched. eg "todo" in a string such as "mastodon".
+     * If the term ends in a non-word character, then \b won't match on
+     * the boundary to the next non-word character, which would likely
+     * be a space. For example `/\bFIX!\b/.test('FIX! blah') === false`.
+     * In these cases, use no bounding match. Same applies for the
+     * prefix, handled below.
+     */
+    const suffix = /\w$/.test(term) ? "\\b" : "";
+
+    if (location === "start") {
+        /*
+         * When matching at the start, ignore leading whitespace, and
+         * there's no need to worry about word boundaries.
+         */
+        prefix = "^\\s*";
+    } else if (/^\w/.test(term)) {
+        prefix = "\\b";
+    } else {
+        prefix = "";
+    }
+
+    return {
+        regexp: new RegExp(prefix + escaped + suffix, "i"),
+        term,
+    };
+}
+
+function walk(ctx: Lint.WalkContext<Options>) {
+    const warningRegExps: RegExpWithTerm[] = ctx.options.terms.map((term) => convertToRegExp(term, ctx.options.location));
+
+    utils.forEachComment(ctx.sourceFile, (fullText, {kind, pos, end}) => {
+        let start = pos + 2;
+        if ((kind !== ts.SyntaxKind.SingleLineCommentTrivia && kind !== ts.SyntaxKind.MultiLineCommentTrivia) ||
+            // exclude empty single-line comments
+            start === end ||
+            // exclude /// <reference path="...">
+            fullText[start] === "/" && ctx.sourceFile.referencedFiles.some((ref) => ref.pos >= pos && ref.end <= end)) {
+            return;
+        }
+        // skip all leading slashes
+        while (fullText[start] === "/") {
```

Comment:
should this rule exclude JSDoc comments? If no, you would need to skip the remaining `*` at the beginning of the text